doc/tutorials/nvmx_virtualization.rst

```diff
@@ -0,0 +1,169 @@
+  on LAPIC passthrough pCPUs
+* Don't support hybrid mode currently
+
+L1 VM configuration
+*******************
+
+The nested virtualization feature is disabled by default in ACRN. You can
+enable it using the :ref:`ACRN configuration toolset <acrn_config_workflow>`
+with the following steps:
+
+- Configure system level features:
+
+   - Edit ``NVMX_ENABLED`` to ``y`` in ACRN scenario XML configuration to
+     enable nested virtualization
+
+   - Edit ``SCHEDULER`` to ``SCHED_NOOP``
+
+   .. code-block:: xml
+      :emphasize-lines: 3, 10
+
+      <FEATURES>
+         <RELOC>y</RELOC>
+         <SCHEDULER>SCHED_NOOP</SCHEDULER>
+         <MULTIBOOT2>y</MULTIBOOT2>
+         <ENFORCE_TURNOFF_AC>y</ENFORCE_TURNOFF_AC>
+         <RDT>
+            <RDT_ENABLED>n</RDT_ENABLED>
+            <CDP_ENABLED>n</CDP_ENABLED>
+         </RDT>
+         <NVMX_ENABLED>y</NVMX_ENABLED>
+         <HYPERV_ENABLED>y</HYPERV_ENABLED>
+         <IOMMU_ENFORCE_SNP>n</IOMMU_ENFORCE_SNP>
+         <ACPI_PARSE_ENABLED>y</ACPI_PARSE_ENABLED>
+         <L1D_VMENTRY_ENABLED>n</L1D_VMENTRY_ENABLED>
+         <MCE_ON_PSC_DISABLED>n</MCE_ON_PSC_DISABLED>
+         <IVSHMEM>
+            <IVSHMEM_ENABLED>n</IVSHMEM_ENABLED>
+            <IVSHMEM_REGION/>
+         </IVSHMEM>
+      </FEATURES>
+
+- Configure system level features:
+
+  - Edit ``cpu_affinity`` to assign pCPU IDs to run the L1 guest. If you are
+    using debug build and need the hypervisor console, please don't assign
+    pCPU0 to the VM that enables nVMX
+  - Edit ``guest_flag`` to enable ``GUEST_FLAG_NVMX_ENABLED`` on the VM that
+    you want to run nested guests on
+  - Edit ``guest_flag`` to enable ``GUEST_FLAG_LAPIC_PASSTHROUGH`` and
+    ``GUEST_FLAG_RT`` on this nVMX enabled VM
+
+   .. code-block:: xml
+      :emphasize-lines: 5,6,9,10,11
+
+      <vm id="0">
+        <vm_type>SOS_VM</vm_type>
+        <name>ACRN SOS VM</name>
+        <cpu_affinity>
+          <pcpu_id>1</pcpu_id>
+          <pcpu_id>2</pcpu_id>
+        </cpu_affinity>
+        <guest_flags>
+          <guest_flag>GUEST_FLAG_NVMX_ENABLED</guest_flag>
+          <guest_flag>GUEST_FLAG_LAPIC_PASSTHROUGH</guest_flag>
+          <guest_flag>GUEST_FLAG_RT</guest_flag>
+        </guest_flags>
+
+
+Launch nested guests
+********************
+
+Run QEMU on L1 guest to launch nested guest. Please note that ``-cpu host``
+is needed to launch nested guests, and currently ``-nographics`` is required
+to be able to run nested guests reliably.
+
+   .. code-block:: bash
+      :emphasize-lines: 2,3,4
+
+      qemu-system-x86_64 \
+        -enable-kvm \
+        -cpu host \
+        -nographic \
+        -m 2G \
+        -smp 2 \
+        -hda /root/ubuntu-20.04.qcow2 \
+        -net nic,macaddr=00:16:3d:60:0a:80 -net tap,script=/etc/qemu-ifup
+
+After nested guest is launched, it is expected to not able to see the nested
+guest from hypervisor ``vcpu_list`` and ``vm_list``.
+
```

Comment:
Do we need add the complete procedure about how to check if the L2 guest is launched successfully?

Reply:
Will, will do.